contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -94,50 +92,50 @@ def is_non_synthetic_python_target(target):
   def is_python_target(target):
     return isinstance(target, PythonTarget)
 
-  def _is_tagged_target(self, target: Target) -> bool:
-    return self.get_options().whitelist_tag_name in target.tags
-
-  def _is_tagged_non_synthetic_python_target(self, target: Target) -> bool:
-    return (self.is_non_synthetic_python_target(target) and
-            self._is_tagged_target(target))
-
-  def _not_tagged_non_synthetic_python_target(self, target: Target) -> bool:
-    return (self.is_non_synthetic_python_target(target) and
-            not self._is_tagged_target(target))
-
-  def _all_targets_are_whitelisted(self, whitelisted_targets: List[Target], all_targets: List[Target]) -> bool:
-    return len(whitelisted_targets) == 0 or len(whitelisted_targets) == len(all_targets)
-
-  def _format_targets_not_whitelisted(self, targets: List[Target]) -> str:
-    output = ''
-    for target in targets:
-      output = output + f"{target.address.spec}\n"
-    return output
-
-  def _whitelist_warning(self, targets_not_whitelisted: List[Target]) -> None:
-    self.context.log.warn(self.WARNING_MESSAGE)
-    if self.get_options().verbose:
-      output = self._format_targets_not_whitelisted(targets_not_whitelisted)
-      self.context.log.warn(f"{output}")
+  def _check_for_untagged_dependencies(
+    self, *, tagged_target_roots: Iterable[Target], tag_name: str
+  ) -> None:
+    untagged_dependencies = {
+      tgt for tgt in Target.closure_for_targets(target_roots=tagged_target_roots)
+      if tag_name not in tgt.tags and self.is_non_synthetic_python_target(tgt)
+    }
+    if not untagged_dependencies:
+      return
+    formatted_targets = "\n".join(tgt.address.spec for tgt in sorted(untagged_dependencies))
+    self.context.log.warn(
+      f"[WARNING]: The following targets are not marked with the tag name `{tag_name}`, "
+      f"but are dependencies of targets that are type checked because they have this tag. You are "
+      f"encouraged to type check these dependencies.\n{formatted_targets}"
```

Comment:
Hm, it's not clear what this means. Will they be skipped, or will this fail?

Reply:
Got it, will reword. These dependencies will be checked because that's how MyPy works. We recommend that they type those dependencies.